crates/trigger-http/src/handler.rs

```diff
@@ -170,13 +171,37 @@ impl HttpHandlerExecutor {
             .data_mut()
             .new_response_outparam(response_tx)?;
 
-        let proxy = Proxy::new(&mut store, &instance)?;
+        enum Handler {
+            Latest(Proxy),
+            Handler2023_10_18(IncomingHandler2023_10_18),
+        }
+
+        let handler = match instance
+            .exports(&mut store)
+            .instance("wasi:http/incoming-handler@0.2.0-rc-2023-10-18")
```

Comment:
Are we sure we want to handle non-"incoming-handler@0.2.0-rc-2023-10-18" interfaces? Our public guarantee is that we only support "incoming-handler@0.2.0-rc-2023-10-18" so we don't necessarily need to support newer versions. If we choose to not support the newer versions as first class, then we won't need to support N number of wasi-http interfaces in the future. We'll only need to make "incoming-handler@0.2.0-rc-2023-10-18" work on the latest version of wasmtime-wasi-http.